ui-react/packages/atlasmap-provider/src/useAtlasmap.tsx

```diff
@@ -1,19 +1,27 @@
 import ky from 'ky';
-import { useEffect, useMemo, useReducer } from 'react';
+import { useCallback, useEffect, useMemo, useReducer } from 'react';
 import { timer } from 'rxjs';
 import { debounce } from 'rxjs/operators';
 import { InspectionType } from './common/config.types';
 import { ConfigModel } from './models/config.model';
 import { DocumentDefinition } from './models/document-definition.model';
-import { ErrorScope, ErrorType, ErrorInfo, ErrorLevel } from './models/error.model';
+import {
+  ErrorScope,
+  ErrorType,
+  ErrorInfo,
+  ErrorLevel,
+} from './models/error.model';
 import { MappingDefinition } from './models/mapping-definition.model';
 import { DocumentManagementService } from './services/document-management.service';
 import { ErrorHandlerService } from './services/error-handler.service';
 import { FieldActionService } from './services/field-action.service';
 import { FileManagementService } from './services/file-management.service';
 import { InitializationService } from './services/initialization.service';
 import { MappingManagementService } from './services/mapping-management.service';
-import { fromDocumentDefinitionToFieldGroup, fromMappingDefinitionToIMappings } from './utils/to-ui-models-util';
+import {
```

Comment:
I think this is where I have a comment/ question.  Essentially all of the component support logic - i.e. all of the business logic  code from:

https://github.com/atlasmap/atlasmap/tree/master/ui/src/app/lib/atlasmap-data-mapper/components

Needs to end up here.  It's great that it's in atlasmap-provider - but how should this be designed?  For example - importAtlasFile() came from toolbar.component.ts.  Should be use the same naming convention as the old UI and import the resulting files?  For example - create MapperUtilsToolbar.ts here in atlasmap-provider to correspond with toolbar.component.ts?

wdyt?

Reply:
All the logic that's presentation specific will have to be implemented in the UI package. All the code that's about providing data to the components will have to be exposed by the provider. Where to put things is up to us, if you think organizing helper code like `importAtlasFile` in something that reminds of the UI component that will use it go for it. I personally don't feel the need for it.

Reply:
Ok - I think that would help organize everything to establish a similar structure - i.e. atlasmap-provider and atlasmap-data-mapper/components. We can always reduce later if it seems over-complex.

Reply:
Sure, works for me